src/test/scala/com/microsoft/hyperspace/index/RefreshIndexTests.scala

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.index
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.catalyst.plans.SQLHelper
+
+import com.microsoft.hyperspace.{Hyperspace, HyperspaceException, SampleData}
+import com.microsoft.hyperspace.util.FileUtils
+
+/**
+ * Unit E2E test cases for RefreshIndex.
+ */
+class RefreshIndexTests extends HyperspaceSuite with SQLHelper {
```

Comment:
Can you add a test case to cover index delete being called on data source where everything was already deleted?

Reply:
This scenario is not supported due to the way `event()` is called for an Action.
Please refer to the other comment above which has a full explanation: https://github.com/microsoft/hyperspace/pull/142#discussion_r489103667

Reply:
I looked at the comment but am not sure why a test case cannot be added (please correct me if I'm wrong). This is a scenario that a user can hit into precisely based on your comment. Why can't the test case elaborate on what's expected in this scenario? (even if that's `Index plan is not valid` - I don't think that's a good error message but we'll talk about this in the other comment)

Reply:
@rapoth What I meant in my first comment here was: 
"If the user deletes ALL source data files and calls refresh, the attempt to refresh will fail and no index refresh will happen". 
Details of the error messages are mentioned in https://github.com/microsoft/hyperspace/pull/142#discussion_r489103667.

I thought you are asking for a positive test case for the scenario (validating a successful refresh if all source data files are deleted). I can add negative test cases to capture both cases I explained in https://github.com/microsoft/hyperspace/pull/142#discussion_r489103667.

Reply:
@rapoth I added a test case to capture both cases under index refresh tests.